frontend/public/components/namespace.jsx

```diff
@@ -531,8 +533,22 @@ export const NamespaceSummary = ({ ns }) => {
       <div className="col-sm-6 col-xs-12">
         {/* Labels aren't editable on kind Project, only Namespace. */}
         <ResourceSummary resource={ns} showLabelEditor={ns.kind === 'Namespace'}>
-          {displayName && <dt>Display Name</dt>}
-          {displayName && <dd>{displayName}</dd>}
+          {<dt>Display Name</dt>}
+          <dd
+            className={classNames({
+              'text-muted': !displayName,
+            })}
+          >
+            {displayName || 'No display name'}
+          </dd>
+          {<dt>Description</dt>}
+          <dd
```

Comment:
We probably want line clamp here as well, even if it's set to a larger value like 10 lines. There's nothing in the API preventing a user from adding up to a 1MB description, which would blow up the layout.

Reply:
I'd add `co-pre-line` as well to preserve line breaks.

Reply:
> We probably want line clamp here as well, even if it's set to a larger value like 10 lines. There's nothing in the API preventing a user from adding up to a 1MB description, which would blow up the layout.

should I set the max lines to 10? cc @lwrigh

Reply:
I would suggest truncating at the 10 line limit like you're suggesting for the description section in the UI. Would it be possible to set a character limit for both the UI and the YAML? That would help to constraint the length of the description text in both views.  

If it isn't an option, we could go the route of truncating the description text and then having it become clickable when the user hovers over it and they can expand it. We could use either a popover or add an expanding/collapsing caret to the description section.

Reply:
@lwrigh There is a collapsible feature in the margin for the YAML editor. I don't think any change is needed there.

Not sure any character limit can be imposed here, but it is an openshift custom resource -- it is possible to think on that but not for 4.6 (as this would be a feature). 

Would you be able to provide a quick mock what you're thinking of for the expand feature? I don't want us going crazy to support this -- again it's post feature freeze so it's limited what we can do here.

Reply:
![Expandable description 1](https://user-images.githubusercontent.com/20227374/90533603-c0fd2f80-e146-11ea-9dd2-c291a373f242.png)
![Expandable description 2](https://user-images.githubusercontent.com/20227374/90533608-c2c6f300-e146-11ea-8668-6df659693310.png)

I was thinking of something like this if we went the caret route.

Reply:
@spadgett ^

Reply:
@lwrigh for the serverless reason text in the table, the suggestion is currently a popover on hover IIRC? Wouldn't we want to have a similar experience in both cases.
Also the description shouldn't be hidden if it's of reasonable size.

Reply:
Yes that's the current suggestion for the reason, We can go the popover route here too. The description here could truncate after 10 lines or so since there's more space around it.